pkg/proxy/proxy.go

```diff
@@ -18,120 +18,137 @@ package proxy
+				nil
+		}
+	}
+
+	if authorized == authorizer.DecisionAllow {
+		return authorized, "", nil
+	}
+
+	return authorizer.DecisionDeny,
+		"No attribute combination matched",
+		nil
+}
+
+func (n *krpAuthorizer) getKubeRBACProxyAuthzAttributes(ctx context.Context, origAttrs authorizer.Attributes) []authorizer.Attributes {
+	u := origAttrs.GetUser()
+	apiVerb := origAttrs.GetVerb()
+	path := origAttrs.GetPath()
 
+	attrs := []authorizer.Attributes{}
 	if n.authzConfig.ResourceAttributes == nil {
 		// Default attributes mirror the API attributes that would allow this access to kube-rbac-proxy
-		allAttrs := append(allAttrs, authorizer.AttributesRecord{
-			User:            u,
-			Verb:            apiVerb,
-			Namespace:       "",
-			APIGroup:        "",
-			APIVersion:      "",
-			Resource:        "",
-			Subresource:     "",
-			Name:            "",
-			ResourceRequest: false,
-			Path:            r.URL.Path,
-		})
-		return allAttrs
+		return append(attrs,
+			authorizer.AttributesRecord{
+				User:            u,
+				Verb:            apiVerb,
+				ResourceRequest: false,
+				Path:            path,
+			})
+
 	}
 
 	if n.authzConfig.Rewrites == nil {
-		allAttrs := append(allAttrs, authorizer.AttributesRecord{
-			User:            u,
-			Verb:            apiVerb,
-			Namespace:       n.authzConfig.ResourceAttributes.Namespace,
-			APIGroup:        n.authzConfig.ResourceAttributes.APIGroup,
-			APIVersion:      n.authzConfig.ResourceAttributes.APIVersion,
-			Resource:        n.authzConfig.ResourceAttributes.Resource,
-			Subresource:     n.authzConfig.ResourceAttributes.Subresource,
-			Name:            n.authzConfig.ResourceAttributes.Name,
-			ResourceRequest: true,
-		})
-		return allAttrs
-	}
+		return append(attrs,
+			authorizer.AttributesRecord{
+				User:            u,
+				Verb:            apiVerb,
+				Namespace:       n.authzConfig.ResourceAttributes.Namespace,
+				APIGroup:        n.authzConfig.ResourceAttributes.APIGroup,
+				APIVersion:      n.authzConfig.ResourceAttributes.APIVersion,
+				Resource:        n.authzConfig.ResourceAttributes.Resource,
+				Subresource:     n.authzConfig.ResourceAttributes.Subresource,
+				Name:            n.authzConfig.ResourceAttributes.Name,
+				ResourceRequest: true,
+			})
 
-	params := []string{}
-	if n.authzConfig.Rewrites.ByQueryParameter != nil && n.authzConfig.Rewrites.ByQueryParameter.Name != "" {
-		if ps, ok := r.URL.Query()[n.authzConfig.Rewrites.ByQueryParameter.Name]; ok {
-			params = append(params, ps...)
-		}
-	}
-	if n.authzConfig.Rewrites.ByHTTPHeader != nil && n.authzConfig.Rewrites.ByHTTPHeader.Name != "" {
-		mimeHeader := textproto.MIMEHeader(r.Header)
-		mimeKey := textproto.CanonicalMIMEHeaderKey(n.authzConfig.Rewrites.ByHTTPHeader.Name)
-		if ps, ok := mimeHeader[mimeKey]; ok {
-			params = append(params, ps...)
-		}
 	}
 
+	params := GetKubeRBACProxyParams(ctx)
 	if len(params) == 0 {
-		return allAttrs
+		return nil
```

Comment:
Hm, this particular case was mentioned by David and I agree, that it would be better to make it fail implicit... but dunno if it is worth introducing an err value in the return, just for this. Maybe with the comment, it makes more sense?

Why is the comment below and not above?

Reply:
I probably had it there as a mental note. I believe the original behavior is kept for now.